darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py

```diff
@@ -396,8 +393,24 @@ def plot_heatmap(
         align="edge",
     )
 
-    # set the x and y tick locations and labels
-    set_x_axis_ticks_and_labels(jointgrid=jgrid, agg_df=agg_df, n_xlabels=4)
+    # retrieve the final DXT segment end time
+    tmax_dxt = float(agg_df["end_time"].max())
+    # calculate the elapsed runtime
+    runtime = report.metadata["job"]["end_time"] - report.metadata["job"]["start_time"]
+    # ensure a minimum runtime value of 1 second
+    runtime = max(runtime, 1)
+    # if the data max time exceeds the runtime, buffer by 1 second
+    # until our timer precision improves to prevent truncation
```

Comment:
I thought we agreed on runtime + 1 second, ignoring DXT data and just passing through to the padded plots. I don't really care too much I don't think if it looks 'ok' when I do a more thorough review, though in cases where DXT heatmap analysis gets automatically suppressed due to size in the future it may not be sustainable.

When we only have to do deal with higher-resolution logs the issue will be less crucial perhaps, though older log files will likely be reality for some time.

Reply:
I guess I figured since we have the option to only pad when necessary it made more sense, but I can switch it back. Either is fine with me

Reply:
Yeah, probably not worth changing at the moment, experience/usage will probably guide future changes anyway.

I think my only substantive design thought at the moment is the encapsulation of some of the new logic before `set_x_axis_ticks_and_labels`. For example, there's a step before the function is called described as:

> rescale x-ticks so x-ticks are adjusted properly downstream

which might logically fit into a function like `set_x_axis_ticks_and_labels()`. But not sure if that is just me trying to get things scoped for convenient testing purposes, and if cramming the rest of that logic in the function would go too far.